src/Microsoft.Azure.ServiceBus/Message.cs

```diff
@@ -61,7 +60,8 @@ public string MessageId
 
             set
             {
-                Message.ValidateMessageId(value);
+                Guard.AgainstNullAndEmpty(nameof(MessageId), value);
```

Comment:
>AgainstNullAndEmpty [](start = 22, length = 19)

Lets change this. I can see scenarios where someone wants to clear the assigned `MessageId`. Lets just guard against null here